### PR TITLE
env prepare explicitly in workflows

### PR DESCRIPTION
**Pull request by ESI-SYD**

### CODE REVIEW

**Comment by chuanqi129** [comment]
For UT test workflow, I saw `No files were found with the provided path: /home/sdp/actions-runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/triton/python/test/unit/language/*.log. No artifacts will be uploaded.`  https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/6229964240. please help to double check whether it artifact a right path

File: .github/scripts/env_prepare.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+if [[ -z "$(pip list | grep torch)" || "$installed_torch_git_version" != "$current_torch_git_version" ]];then
+    echo -e "========================================================================="
+    echo "Public torch BUILD"
+    echo -e "========================================================================="
+    rm -rf ${HOME}/${JOB_WORKSPACE}/pytorch
+    rm -rf ${HOME}/${JOB_WORKSPACE}/intel-extension-for-pytorch
+    cd ${HOME}/${JOB_WORKSPACE}
+    pip uninstall torch -y
+    git clone -b ${TORCH_BRANCH} ${TORCH_REPO}
+    git clone -b xpu-master https://github.com/intel/intel-extension-for-pytorch
```
[the remark this answers is not in the document]
Comment (ESI-SYD) [reply .github/scripts/env_prepare.sh]: updated

File: .github/workflows/triton_xpu_backend_ci.yml [hunk 2]
```diff
@@ -211,4 +245,4 @@ jobs:
         uses: actions/upload-artifact@v3
         with:
           name: Triton-ci-data
-          path: /home/sdp/actions-runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/triton/python/test/unit/language/*.log
+          path: /home/sdp/triton-preci/triton/python/test/unit/language/*.log
```
Comment (chuanqi129) [inline .github/workflows/triton_xpu_backend_ci.yml:248]: I guess UT nightly also need such change @ESI-SYD

Comment (ESI-SYD) [reply .github/workflows/triton_xpu_backend_ci.yml:248]: Checked with ruijie, UT nightly use default workspace, not change to use triton-preci or triton-nightly like other workflows

**Review by chuanqi129 — APPROVED** [review]

**Pull request merged**